File: babydata/src/main/java/com/zero/babydata/entity/BabyInfo.java
package com.zero.babydata.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.text.SimpleDateFormat;
import java.util.Date;

@Entity(tableName = "Baby_Info")
public class BabyInfo {
    @PrimaryKey(autoGenerate = true)
    public int babyId;

    public String name;
    public String gender;
    public Long birthDate;
    public float birthWeight;
    public float birthHeight;
    public String bloodType;  // 新增字段：血型
    public String extra;  // 新增字段：血型


    public String generateCurrentData(){
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
        String currentDateAndTime = sdf.format(new Date());
        return currentDateAndTime;
    }
}

File: babydata/src/main/java/com/zero/babydata/entity/BabyInfo.kt
package com.zero.babydata.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "Baby_Info")
data class BabyInfo(
    @PrimaryKey(autoGenerate = true)
    var babyId: Int = 0,
    var name: String = "",
    var gender: String = "",
    var birthDate: Long = 0L,
    var birthWeight: Float = 0f,
    var birthHeight: Float = 0f,
    var bloodType: String = "",
    var extra: String = ""
)
